src/erlang_ls.erl

```diff
@@ -19,16 +28,54 @@ parse_args(["--transport", Name | Rest]) ->
                 "tcp"   -> els_tcp;
                 "stdio" -> els_stdio
               end,
-  application:set_env(erlang_ls, transport, Transport),
+  application:set_env(?APP, transport, Transport),
   parse_args(Rest);
 parse_args(["--port", Port | Rest]) ->
-  application:set_env(erlang_ls, port, list_to_integer(Port)),
+  application:set_env(?APP, port, list_to_integer(Port)),
   parse_args(Rest);
 %% For backward compatibility with clients
 parse_args([Port | Rest]) ->
-  application:set_env(erlang_ls, port, list_to_integer(Port)),
+  application:set_env(?APP, port, list_to_integer(Port)),
   parse_args(Rest).
 
+%%==============================================================================
+%% Lager configuration
+%%==============================================================================
+
+-spec lager_config() -> ok.
+lager_config() ->
+  LogRoot  = log_root(),
+  Handlers = case application:get_env(?APP, lager_handlers) of
+               {ok, Value} -> Value;
+               undefined   -> default_lager_handlers(LogRoot)
+             end,
+  ok = application:load(lager),
+  ok = application:set_env(lager, handlers, Handlers),
+  ok = application:set_env(lager, crash_log, "crash.log"),
+  ok = application:set_env(lager, log_root, LogRoot),
+  ok.
+
+-spec default_lager_handlers(string()) -> [any()].
+default_lager_handlers(LogRoot) ->
+  LogFile = filename:join([LogRoot, "info.log"]),
+  ok      = filelib:ensure_dir(LogFile),
+  [ { lager_file_backend
+    , [ {file, LogFile}
+      , {level, info}
+      ]
+    }
+  ].
+
+-spec log_root() -> string().
+log_root() ->
+  {ok, CurrentDir} = file:get_cwd(),
+  Dirname = filename:basename(CurrentDir),
+  filename:join([ "/usr/local/var/log/erlang_ls", Dirname]).
```

Comment:
I would like this to be configurable. Also, Isn't `/var/log` enough?

Reply:
Configurable through the application's env vars?